zulip_bots/zulip_bots/bots/merels/test_merels.py

```diff
@@ -124,3 +158,46 @@ def test_message_handler_helpers(self) -> None:
             bot.game_message_handler.alert_move_message("foo", "move 1,1"),
             "foo :move 1,1",
         )
+
+    def test_move_after_join_invokes_make_move_and_replies(self) -> None:
+        """
+        Start a two-player game, send a move via the adapter, count make_move calls,
+        and assert we get a reply. Try both users to avoid turn assumptions.
+        """
+        import types
+
+        bot, bot_handler = self._get_handlers()
+
+        # Start 2P game.
+        _ = self.send_and_collect(
+            bot, bot_handler, "start game", user="foo@example.com", user_name="foo"
+        )
+        _ = self.send_and_collect(bot, bot_handler, "join", user="bar@example.com", user_name="bar")
+
+        # Count model.make_move invocations.
+        self.assertTrue(hasattr(bot.model, "make_move"), "Merels model has no make_move method")
+        original = bot.model.make_move
+        calls = {"n": 0}
+
+        def _wrapped_make_move(*args, **kwargs):
+            calls["n"] += 1
+            return original(*args, **kwargs)
+
+        bot.model.make_move = types.MethodType(_wrapped_make_move, bot.model)  # type: ignore[attr-defined]
```

Comment:
Can you explain why you did this weird mocking strategy here? It seems very hacky and I'm not sure what you're actually testing as a result. I think the goal here would usually be to just verify that the output of the actual bot's `make_move` function.